src/parser/Abstract_Class_Parser.i.hh

```diff
@@ -14,6 +14,36 @@
 #ifndef utils_Abstract_Class_Parser_i_hh
 #define utils_Abstract_Class_Parser_i_hh
 
+//---------------------------------------------------------------------------//
+template <typename Abstract_Class, typename Context,
+          Context const &get_context()>
+Contextual_Parse_Functor<Abstract_Class, Context, get_context>::
+    Contextual_Parse_Functor(SP<Abstract_Class> parse_function(Token_Stream &,
+                                                               Context const &))
+    : f_(parse_function) {}
+
+template <typename Abstract_Class, typename Context,
+          Context const &get_context()>
+SP<Abstract_Class>
+Contextual_Parse_Functor<Abstract_Class, Context, get_context>::
+operator()(Token_Stream &tokens) const {
+  return f_(tokens, get_context());
+}
+
+//===========================================================================//
+/*!
+ * Helper class defining a table of raw strings created by strdup that will be properly
+ * deallocated using free on program termination.
+ */
+class c_string_vector {
+public:
+  ~c_string_vector();
+  c_string_vector(void) : data(0) { /* empty */
+  }
+  vector<char *> data;
+};
+extern c_string_vector abstract_class_parser_keys;
```

Comment:
Kent - I forgot to submit this this request last night (it was still open in my web browser this morning).  This one-liner is also required to get pr140 to work on windows. Sorry for the omission.
```diff
-extern c_string_vector abstract_class_parser_keys;
+DLL_PUBLIC_parser extern c_string_vector abstract_class_parser_keys;
```